spec/models/mailing_list/signup_spec.rb

```diff
@@ -0,0 +1,210 @@
+        it { is_expected.to allow_value(" me@you.com ").for(:email) }
+        it { is_expected.not_to allow_value("me@you").for(:email).with_message("Enter a valid email address") }
+      end
+    end
+
+    describe "#first_name" do
+      it { is_expected.to validate_presence_of(:first_name) }
+      it { is_expected.to validate_length_of(:first_name).is_at_most(256) }
+    end
+
+    describe "#last_name" do
+      it { is_expected.to validate_presence_of(:last_name) }
+      it { is_expected.to validate_length_of(:last_name).is_at_most(256) }
+    end
+
+    describe "#channel_id" do
+      let(:options) { channels.map(&:id) }
+      it { is_expected.to allow_values(options).for(:channel_id) }
+      it { is_expected.to allow_value(nil, "").for(:channel_id) }
+      it { is_expected.not_to allow_value(12_345).for(:channel_id) }
+    end
+
+    describe "#preferred_teaching_subject_id" do
+      let(:teaching_subjects) { GetIntoTeachingApiClient::Constants::TEACHING_SUBJECTS }
+      let(:message) { "Choose a subject from the list" }
+
+      it { is_expected.to validate_presence_of(:preferred_teaching_subject_id).with_message(message) }
+
+      describe "values" do
+        specify "allows teaching subject values" do
+          teaching_subjects.each_value do |ts|
+            expect(subject).to allow_value(ts).for(:preferred_teaching_subject_id)
+          end
+        end
+
+        specify "doesn't allow other values" do
+          %w[abc 123].each do |invalid|
+            expect(subject).not_to allow_value(invalid).for(:preferred_teaching_subject_id).with_message(message)
+          end
+        end
+      end
+    end
+
+    describe "#consideration_journey_stage_id" do
+      let(:journey_stages) { GetIntoTeachingApiClient::PickListItemsApi.new.get_candidate_journey_stages }
+      let(:message) { "Choose an option from the list" }
+
+      it { is_expected.to validate_presence_of(:consideration_journey_stage_id).with_message(message) }
+
+      describe "values" do
+        specify "allows teaching subject values" do
+          journey_stages.map(&:id).each do |js|
+            expect(subject).to allow_value(js).for(:consideration_journey_stage_id)
+          end
+        end
+
+        specify "doesn't allow other values" do
+          %w[abc 123].each do |invalid|
+            expect(subject).not_to allow_value(invalid).for(:consideration_journey_stage_id).with_message(message)
+          end
+        end
+      end
+    end
+
+    describe "#degree_status_id" do
+      let(:degree_statuses) { GetIntoTeachingApiClient::PickListItemsApi.new.get_qualification_degree_status }
+      let(:message) { "Select your degree stage from the list" }
+
+      it { is_expected.to validate_presence_of(:degree_status_id).with_message(message) }
+
+      describe "values" do
+        specify "allows teaching subject values" do
+          degree_statuses.map(&:id).each do |ds|
+            expect(subject).to allow_value(ds).for(:degree_status_id)
```

Comment:
You can pass multiple values to these shoulda matchers with `allow_values` to make them more concise:

`it { is_expected.to allow_values(degree_statuses.map(&:id)).for(:degree_status_id) }`